### PR TITLE
bump: Samples to akka-projection 1.4.0-M4, akka-dependencies 23.04.0-M1

### PR DESCRIPTION
**Pull request by johanandren**

### CODE REVIEW

File: samples/grpc/shopping-cart-service-scala/build.sbt [hunk 1]
```diff
@@ -31,7 +31,7 @@ val AkkaHttpVersion = "10.5.0"
 val AkkaManagementVersion = "1.3.0"
 val AkkaPersistenceR2dbcVersion = "1.1.0-M8"
 val AkkaProjectionVersion =
-  sys.props.getOrElse("akka-projection.version", "1.4.0-M3-124-96b9d6c2-SNAPSHOT") // FIXME
+  sys.props.getOrElse("akka-projection.version", "1.4.0-M4") // FIXME
 val AkkaDiagnosticsVersion = "2.0.0"
```
Comment (patriknw) [inline samples/grpc/shopping-cart-service-scala/build.sbt:36]: what about the RES samples?

**Review by patriknw — COMMENTED** [review]
Thanks, I'll update the Java samples with the missing doc snippets for the event filtering

File: samples/grpc/shopping-cart-service-scala/build.sbt (hunk 1 above)
Comment (johanandren) [reply samples/grpc/shopping-cart-service-scala/build.sbt:36]: Missed those! Pushed now.

Comment (johanandren) [reply samples/grpc/shopping-cart-service-scala/build.sbt:36]: Hmmm what. Pushed but github doesn't show it.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**